src/main/java/com/maddyhome/idea/vim/extension/replacewithregister/ReplaceWithRegister.kt

```diff
@@ -148,6 +149,22 @@ private fun doReplace(editor: Editor, context: DataContext, caret: ImmutableVimC
 
   var usedType = savedRegister.type
   var usedText = savedRegister.text
+
+  val selection = visualSelection.caretsAndSelections.values.first()
```

Comment:
This logic is incorrect for the multicaret scenarios. This function is started for different carets, but we get the `first()` value for each call.